source/iterator/snapshot.go

```diff
@@ -0,0 +1,235 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/conduitio-labs/conduit-connector-vitess/columntypes"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/proto/query"
+	"vitess.io/vitess/go/vt/vtgate/vtgateconn"
+
+	"github.com/doug-martin/goqu/v9"
+	// we need the import to work with the mysql dialect.
+	_ "github.com/doug-martin/goqu/v9/dialect/mysql"
+)
+
+// Snapshot is an implementation of a Snapshot iterator for Vitess.
+type Snapshot struct {
+	session *vtgateconn.VTGateSession
+	records chan sdk.Record
```

Comment:
Would be good to mention a bit more about the channel (why do we actually need it, why not a slice, why is it buffered, etc.)

Reply:
I've added a comment explaining the purpose of this field.